sbpy/spectroscopy/core.py

```diff
@@ -9,41 +9,262 @@
+                                                    (k*300 * u.K).value) < 1:
+
+        if df == 0 or 2:
+
+            intl = lgint*(300*u.K / temp)**(2)*np.exp(-(1/temp - 1/(300*u.K))
+                                                      * elo_J/k)
+
+        else:
+
+            intl = lgint*(300*u.K/temp)**(5/2)*np.exp(-(1/temp - 1/(300*u.K))
+                                                      * elo_J/k)
+
+    else:
+
+        intl = lgint*(part300/partition)*(np.exp(-elo_J/(k*temp)) -
+                                          np.exp(-energy_J/(k*temp))) / \
+               (np.exp(-elo_J/(k*300 * u.K)) - np.exp(-energy_J/(k*300*u.K)))
+
+    return intl
+
+
+def einstein_coeff(temp_estimate, transition_freq, mol_tag):
 
-def einstein_coeff(frequency):
     """
     Einstein coefficient from molecular data
 
     Parameters
     ----------
-    frequency : `~astropy.units.Quantity`
-        Transition frequency
+    transition_freq : `~astropy.units.Quantity`
+        Transition frequency in MHz
+
+    temp_estimate : `~astropy.units.Quantity`
+        Estimated temperature in Kelvins
+
+    mol_tag : int or str
+        Molecule identifier. Make sure it is an exclusive identifier.
 
     Returns
     -------
-    einstein_coeff : float
+    einstein_coeff : `~astropy.units.Quantity`
         Spontaneous emission coefficient
 
-    not implemented
     """
 
+    mol_data = molecular_data(temp_estimate, transition_freq, mol_tag)
+
+    t_freq = mol_data[0]
+    temp = mol_data[1]
+    lgint = mol_data[2]
+    part300 = mol_data[3]
+    partition = mol_data[4]
+    gu = mol_data[5]
+    energy_J = mol_data[6]
+    elo_J = mol_data[7]
+
+    h = con.h.to('J*s')  # planck constant
 
-def total_number(integrated_flux, frequency):
+    k = con.k_B.to('J/K')  # boltzman constant
+
+    intl = intensity_conversion(temp_estimate, transition_freq, mol_tag)
+
+    if (h*t_freq/(k*temp)).decompose().value and \
+            (h*t_freq/(k*300*u.K)).decompose().value < 1:
+
+        au = (lgint*t_freq
+              * (part300/gu)*np.exp(energy_J / (k*300*u.K))*(1.748e-9)).value
+
+    else:
+
+        au = (intl*(t_freq)**2 *
+              (partition/gu)*(np.exp(-(elo_J/(k*temp)).value) -
+                              np.exp(-(energy_J/(k*temp)).value))**(-1)
+              * (2.7964e-16)).value
+
+    return au / u.s
+
+
+def photod_rate(time, time_scale, target, id_type, observatory, format,
+                mol_tag):
+
+    epoch = Time(time, scale=time_scale, format=format)
+    obj = Horizons(id=target, epochs=epoch.jd, location=observatory,
+                   id_type=id_type)
+
+    try:
+        orb = obj.ephemerides()
+
+    except ValueError:
+
+        raise
+
+    delta = (orb['delta'].data * u.au).to('m')
+    r = (orb['r'].data)
+    cat = JPLSpec.get_species_table()
+    mol = cat[cat['TAG'] == mol_tag]
+    name = mol['NAME'].data[0]
+
+    timescale = photo_timescale(name)
+
+    if type(timescale) == np.array:
```

Comment:
use `isinstance` for checking the type

Reply:
I did this not to assert the type, it's because timescale could also be an array in the case of quiet vs active Sun, and in that case we would have to pull the first item of the array otherwise the rest of the calculation for delta will get an error because it treats the timescale as a value and not an array.

Reply:
Good catch! If this works it is OK, generally I think that isinstance is preferred over type because it takes care of inheritance. Also the type should be `np.ndarray`

Reply:
It seems that photo_timescale always return a Quantity (a subclass of np.ndarray)

```
type([3.15e5, 1.35e5] * u.s) == np.ndarray
False
isinstance([3.15e5, 1.35e5] * u.s, np.ndarray)
True
isinstance(5.0e5 * u.s, np.ndarray)
True
```

checking type equality does not work, and even a one element Quantity will be a subclass of np.ndarray. @mkelley do you have any idea how to check the output of photo_timescale?

Reply:
I suggest you do:

```
if len(timescale.shape) == 0:
    # timescale is scalar
else:
    # timescale has at least 1 dimension
```

Reply:
That works.  I probably would have used `.ndim`:
```
if timescale.ndim == 0:
    # scalar
else:
    # array
```